### PR TITLE
Delete invalid settings for system indices

### PR DESCRIPTION
**Pull request by grcevski**
When upgrading from 7 to 8 we archive any index settings that are invalid or unknown, allowing the end users to review them and delete them after confirming their configuration is still correct after the update. However, this same logic applies to system indices, where the end-users cannot change anything.

Leaving archived settings behind on system indices renders the cluster unusable in some situations, without ability for end users to remove the invalid settings.

With this PR we delete unknown/invalid settings on system indices, while archiving as usual on regular indices.

Relates to #88324

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java [hunk 1]
```diff
@@ -206,26 +206,50 @@ public Set<Entry<String, NamedAnalyzer>> entrySet() {
      * Identify invalid or unknown index settings and archive them. This leniency allows Elasticsearch to load
```
Comment (williamrandolph) [inline server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifier.java:206]: We should update this javadoc too.

File: server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java [hunk 2]
```diff
@@ -956,6 +956,51 @@ public Settings archiveUnknownOrInvalidSettings(
...
+     */
+    public Settings deleteUnknownOrInvalidSettings(
+        final Settings settings,
+        final Consumer<Map.Entry<String, String>> unknownConsumer,
+        final BiConsumer<Map.Entry<String, String>, IllegalArgumentException> invalidConsumer
+    ) {
+        Settings.Builder builder = Settings.builder();
+        boolean changed = false;
+        for (String key : settings.keySet()) {
+            try {
```
Comment (williamrandolph) [inline server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java:979]: It looks to me like line 982 is what we're relying on to throw the IllegalArgumentException. Would it make sense to scope this try/catch block more tightly, or to add a comment on that line that `setting.get(settings)` will throw an IllegalArgumentException if a setting has an invalid value?

File: server/src/test/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifierTests.java [hunk 3]
```diff
@@ -26,26 +26,61 @@ public class IndexMetadataVerifierTests extends ESTestCase {
...
         assertNotSame(indexMetadata, src);
         assertEquals("-1", indexMetadata.getSettings().get("archived.index.refresh.interval"));
 
         src = newIndexMeta("foo", indexMetadata.getSettings()); // double archive?
-        indexMetadata = service.archiveBrokenIndexSettings(src);
+        indexMetadata = service.archiveOrDeleteBrokenIndexSettings(src);
+        assertSame(indexMetadata, src);
+    }
+
+    public void testArchiveBrokenSystemIndexSettings() {
```
Comment (williamrandolph) [inline server/src/test/java/org/elasticsearch/cluster/metadata/IndexMetadataVerifierTests.java:52]: nit: `testDeleteBrokenSystemIndexSettings()` ?

**Review by williamrandolph — APPROVED** [review]
A few nitpicks, but otherwise LGTM!

File: server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java (hunk 2 above)
Comment (grcevski) [reply server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java:979]: Yeah, I was following what was done for archive. I'll add a related comment.

**Pull request merged**